### PR TITLE
Changed link on concurrency page

### LINKED ISSUE #4408: Change 'error prone` link in 'Concurrency in Dart > How isolates work' to data race definition
### Page URL

https://dart.dev/guides/language/concurrency#how-isolates-work

### Page source

https://github.com/dart-lang/site-www/blob/main/src/language/concurrency/index.md

### Describe the problem

In this section:

> Most modern devices have multi-core CPUs. To take advantage of all those cores, developers sometimes use shared-memory threads running concurrently. However, shared-state concurrency is [error prone](https://en.wikipedia.org/wiki/Race_condition#In_software) and can lead to complicated code.

I think it would make more sense here for "error prone" to link to an explanation about _data races_ not _race conditions_. 

Not sharing state between isolates prevents _data races_ from happening since _data races_ by definition require shared state in order to occur.

You can still have _race conditions_ even without shared state, and isolates do not eliminate _race conditions_.

### Expected fix

Link to an explanation about _data races_ instead of _race conditions_. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by atsansone**
Fixes #4408

### CODE REVIEW

**Review by MaryaBelanger — CHANGES_REQUESTED** [review]
I'm inclined to go with Paul's comment saying the original link is better. And then, as the user said, instead change the wording so it's clearer:

> the use of isolates doesn't prevent the user's code from experiencing race conditions, it just prevents data races.
> 
> I just want this point to be communicated clearly in the docs.

That clarification wouldn't really make sense to add where the link in question is actually mentioned, since isolates aren't even introduced at that point, but maybe a few lines down after 176.

I can take over the PR and implement this, just lmk or reassign me.

**Comment by atsansone** [comment]
@MaryaBelanger : Take it. No issues here.

**Comment by MaryaBelanger** [comment]
@stereotype441 this is a very small change, but I'd appreciate it if you could confirm the wording is still technically correct (lifted entirely from #4408)

File: src/language/concurrency/index.md [hunk 1]
```diff
@@ -167,18 +167,21 @@ visit the [asynchronous programming codelab][].
 ## How isolates work
 
 Most modern devices have multi-core CPUs.
-To take advantage of all those cores,
+To take advantage multiple cores,
 developers sometimes use shared-memory threads running concurrently.
 However, shared-state concurrency is
-[error prone](https://en.wikipedia.org/wiki/Race_condition#In_software) and
+[error prone](https://en.wikipedia.org/wiki/Race_condition#Data_race) and
```
[the remark this answers is not in the document]
Comment (MaryaBelanger) [reply src/language/concurrency/index.md:173]: Thanks Paul! Opinions help a lot :)

Comment (MaryaBelanger) [inline src/language/concurrency/index.md:173]:
```suggestion
[error prone](https://en.wikipedia.org/wiki/Race_condition#In_software) and
```

**Comment by atsansone** [comment]
@MaryaBelanger : Made a slight change off your change. Have a look. I hope we can approve and merge soon.

**Review by MaryaBelanger — APPROVED** [review]
@atsansone Thanks for fixing that parenthetical!

**Comment by mmcdon20** [comment]
I'm satisfied with these changes. Thanks @MaryaBelanger, @atsansone, @stereotype441.

**Pull request merged**